packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/resolve.rs

```diff
@@ -0,0 +1,112 @@
+                            .await
+                            .change_context(ResolveError::StoreReadError)?
+                            .resolve(tail_path_segments, context)
+                            .await;
+                    }
+                    "properties" => return self.entity.resolve(tail_path_segments, context).await,
+                    "outgoingLinks" => {
+                        // TODO: Use relation tables
+                        //   see https://app.asana.com/0/0/1202884883200942/f
+                        return Ok(Literal::List(
+                            context
+                                .read_active_links_by_source(self.id)
+                                .await
+                                .change_context(ResolveError::StoreReadError)?
+                                .then(|link| async {
+                                    link.change_context(ResolveError::StoreReadError)?
+                                        .resolve(tail_path_segments, context)
+                                        .await
+                                })
+                                .try_collect()
+                                .await?,
+                        ));
+                    }
+                    "incomingLinks" => {
+                        // TODO: Use relation tables
+                        //   see https://app.asana.com/0/0/1202884883200942/f
+                        return Ok(Literal::List(
+                            context
+                                .read_active_links_by_target(self.id)
+                                .await
+                                .change_context(ResolveError::StoreReadError)?
+                                .then(|link| async {
+                                    link.change_context(ResolveError::StoreReadError)?
+                                        .resolve(tail_path_segments, context)
+                                        .await
+                                })
+                                .try_collect()
+                                .await?,
+                        ));
+                    }
+                    _ => Literal::Null,
+                };
+
+                if tail_path_segments.is_empty() {
+                    Ok(literal)
+                } else {
+                    literal.resolve(tail_path_segments, context).await
+                }
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl<C> Resolve<C> for Entity
+where
+    C: Sync + ?Sized,
+{
+    async fn resolve(&self, path: &[PathSegment], context: &C) -> Result<Literal, ResolveError> {
+        match path {
+            [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+            [head_path_segment, tail_path_segments @ ..] => {
+                let uri = BaseUri::new(head_path_segment.identifier.clone())
+                    .into_report()
+                    .change_context(ResolveError::StoreReadError)?;
```

Comment:
Why is this a `StoreReadError`?

Reply:
I "fixed" this in 6d03a3fe953e3a37bbb69766fd0619b3d9117f76, however, `ResolveError` is badly designed, instead it should be separated structs as base errors. This was an oversight when cleaning up after initially implementing the queries. You want me to change this in this PR or in a follow up?

Reply:
In the interest of progress 💪 lets do it in a follow-up 🙃